src/NamedItemContainer.js

```diff
@@ -17,73 +17,53 @@ class NamedItemContaner {
   }
 
   async getNamedItems() {
-    try {
-      const client = await this._oneDrive.getClient();
-      const result = await client.get(`${this.uri}/names`);
-      return result.value.map((v) => ({
-        name: v.name,
-        value: v.value,
-        comment: v.comment,
-      }));
-    } catch (e) {
-      this.log.error(StatusCodeError.getActualError(e));
-      throw new StatusCodeError(e.message, e.statusCode || 500);
-    }
+    const result = await this._oneDrive.doFetch(`${this.uri}/names`);
+    return result.value.map((v) => ({
+      name: v.name,
+      value: v.value,
+      comment: v.comment,
+    }));
   }
 
   async getNamedItem(name) {
     try {
-      const client = await this._oneDrive.getClient(false);
-      return await client.get(`${this.uri}/names/${name}`);
+      return this._oneDrive.doFetch(`${this.uri}/names/${name}`);
     } catch (e) {
       if (e.statusCode === 404) {
         return null;
       }
-      this.log.error(StatusCodeError.getActualError(e));
-      throw new StatusCodeError(e.message, e.statusCode || 500);
+      throw e;
     }
   }
 
   async addNamedItem(name, reference, comment) {
     try {
-      const client = await this._oneDrive.getClient();
-      await client({
-        uri: `${this.uri}/names/add`,
+      return this._oneDrive.doFetch(`${this.uri}/names/add`, false, {
```

Comment:
did you remove the `await` here intentionally? we kept it here, so that the exception from the `get()` was cought here.

Reply:
Frankly It was not obvious to me why you would need to await the return value in an async function. 

The following might be cleaner:
```
  getNamedItem(name) {
    return this._oneDrive.doFetch(`${this.uri}/names/${name}`)
      .catch((err) => {
        if (e.statusCode === 404) {
          return null;
        }
        throw e;
      }
    );
  }
```